File: nextcart_backend/src/main/java/com/mhjoy/nextcart/product/service/BrandService.java
package com.mhjoy.nextcart.product.service;

import com.mhjoy.nextcart.product.dto.request.CreateBrandRequestDto;
import com.mhjoy.nextcart.product.dto.request.UpdateBrandRequestDto;
import com.mhjoy.nextcart.product.dto.response.BrandResponseDto;

import java.util.List;

public interface BrandService {

    BrandResponseDto createBrand(CreateBrandRequestDto requestDto);

    BrandResponseDto updateBrand(Long brandId, UpdateBrandRequestDto requestDto);

    void deleteBrand(Long brandId);

    List<BrandResponseDto> getBrands();
}

File: nextcart_admin_frontend/src/services/brandService.js
import apiClient from './apiClient'
import { ENDPOINTS } from '@/constants/api'

const brandService = {
  getBrands: async () => {
    const response = await apiClient.get(ENDPOINTS.BRANDS)
    return response.data ?? []
  },

  createBrand: async (payload) => {
    const response = await apiClient.post(ENDPOINTS.BRANDS, payload)
    return response.data
  },

  updateBrand: async (brandId, payload) => {
    const response = await apiClient.put(`${ENDPOINTS.BRANDS}/${brandId}`, payload)
    return response.data
  },

  deleteBrand: async (brandId) => {
    const response = await apiClient.delete(`${ENDPOINTS.BRANDS}/${brandId}`)
    return response.data
  },
}

export default brandService
